### ISSUE TITLE
How can I add tooltip on legend truncated labels?

### ISSUE THREAD

**Issue by sagar-vaghela** [amcharts/amcharts5#1796]
![image](https://github.com/user-attachments/assets/b38bb337-b9a8-406a-b9f9-de91d26d679b)
```JavaScript
if (isLegendVisible) {
      const legendRoot = am5.Root.new(legendContainerRef.current);
      setChartThemes(legendRoot, themeMode === 'dark');

      const tooltip = am5.Tooltip.new(legendRoot, {
        pointerOrientation: 'vertical',
        getFillFromSprite: false,
        getStrokeFromSprite: false,
        dy: 20,
        dx: -20,
      });

      customizeTooltip(themeMode === 'dark', tooltip);

      const legend = legendRoot.container.children.push(
        am5.Legend.new(legendRoot, {
          centerX: am5.percent(50),
          x: am5.percent(50),
          useDefaultMarker: true,
          layout: am5.GridLayout.new(legendRoot, {
            maxColumns: 3,
            // fixedWidthGrid: true,
          }),
          tooltip: tooltip,
          paddingBottom: 50,
        }),
      );

      legend.itemContainers.template.setAll({
        tooltipText: '{name}',
      });

      legend.labels.template.setAll({
        fontFamily: 'Open Sans,Roboto,sans-serif',
        fontWeight: '400',
        fontSize: 16,
        maxWidth: 160,
        ellipsis: '...',
        oversizedBehavior: 'truncate',
        tooltipText: '{name}',
        interactive: true,
      });

      legend.markers.template.setup = (marker) => {
        if (!legendRoot) return;
        const check = am5.Graphics.new(legendRoot, {
          fill: am5.color(0x000000),
          fillOpacity: 1,
          layer: 40,
          svgPath:
            'M15.75 2.527c-.61-.468-1.46-.328-1.902.32l-6.325 9.255L4.04 8.328a1.3 1.3 0 0 0-1.922-.062 1.505 1.505 0 0 0-.062 2.043s4.234 4.695 4.843 5.168c.61.468 1.457.328 1.903-.32L16.05 4.55c.445-.653.308-1.555-.301-2.024Zm0 0',
        });

        check.states.create('disabled', {
          fillOpacity: 0,
        });

        marker.children.push(check);
      };

      legend.events.on('boundschanged', function () {
        if (legendContainerRef.current) {
          legendContainerRef.current.style.height = legend.height() + 'px';
        }
      });

      // Track toggled state
      legend.itemContainers.template.events.on('click', function (ev) {
        const item = ev.target.dataItem?.dataContext as am5xy.XYSeries;
        const name = item.get('name');
        if (item.get('visible')) {
          if (name) {
            setLegendState((prevState) => Array.from(new Set([...prevState, name])));
          }
        } else {
          // remove toggled on series from state
          setLegendState((prevState) => prevState.filter((series) => series !== name));
        }
      });

      legendRoot._logo?.dispose();
      legend.data.setAll(chart.series.values);
      legendRootRef.current = legendRoot;
    }
```
    
  Currently, the implementation displays tooltips for all legends, even when they are not truncated. Additionally, I have to manually adjust the tooltip's position and add padding at the bottom of the root element to ensure that the tooltip is visible.

**Comment by martynasma** [comment]
There's currently no way to determine from a user code if the label was truncated as it happens on renderer level.

We will consider adding something, and will get back to you. I can't provide an ETA, though.

**Comment by martynasma** [comment]
Implemented in [5.10.9](https://github.com/amcharts/amcharts5/releases/tag/5.10.9).

## [5.10.9] - 2024-12-20

### Added
- Added way to determine if text label was autotruncated (`label.text._display.trunacted`).
- Heikin Ashi indicator added to `StockChart`.

### Changed
- The following settings will carry over to `Label`'s HTML content if set: `fontFamily`, `fontSize`, `fontStyle`, `fontWeight`, `fontStyle`, `fontVariant`, `textDecoration`.
- When an `MapChart` is zoomed using mouse wheel it will no longer stop page scroll when its fully zoomed out.
- When arranging tooltips on an `XYChart`, will now pay attention to `tooltipContainerBounds.bottom` setting.

### Fixed
- `ZoomableContainer` was not zooming out properly if `minZoomLevel` was `< 1`.
- `PicturePattern` was ignoring `colorOpacity` setting.
- `PicturePattern` with `fit: "pattern"` set would sometimes not show actual image.
- "Measure" drawing tool of `StockChart` was not including one volume bar value.
- If a timezone was set on a chart and browser's locale was set to some non-English language, weekly grid was shown incorrecly on a `DateAxis`.
- Changing `text` on a `Label` with `oversizedBehavior="fit"` would sometimes not revert to original scale even if the new text would fit into max width.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
